#!/usr/bin/env python
PACKAGE = "gate_detection"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("canny_threshold1",       int_t,  0,  "threshold 1 for the hypertesis",   100,    0,  1000)
gen.add("canny_threshold2",       int_t,  0,  "threshold 2 for the hypertesis",   200,    0,  1000)

canny_aperture_size_enum = gen.enum([ gen.const("Small",      int_t, 3, "A small constant"),
                       gen.const("Medium",     int_t, 5, "A medium constant"),
                       gen.const("Large",      int_t, 7, "A large constant")],
                     "An enum to set sobel procedure aperture size")

gen.add("canny_aperture_size", int_t, 3, "A size parameter which is edited via an enum", 3, 3, 7, edit_method=canny_aperture_size_enum)

exit(gen.generate(PACKAGE, "gate_detection_cfg", "GateDetection"))